pkg/registry/extensions/daemonset/strategy_test.go

```diff
@@ -31,3 +37,71 @@ func TestDefaultGarbageCollectionPolicy(t *testing.T) {
 		t.Errorf("DefaultGarbageCollectionPolicy() = %#v, want %#v", got, want)
 	}
 }
+
+func TestSelectorImmutability(t *testing.T) {
+	tests := []struct {
+		method                 string
+		url                    string
+		oldSelectorLabels      map[string]string
+		newSelectorLabels      map[string]string
+		expectedAPIGroup       string
+		expectedAPIVersion     string
+		expectedSelectorLabels map[string]string
+	}{
+		{"GET", "/apis/apps/v1beta2/namespaces", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "apps", "v1beta2", map[string]string{"a": "b"}},
```

Comment:
To make the URLs more realistic, use something like `"/apis/apps/v1beta2/namespaces/default/daemonsets/dsname"`.